lib/evinser/evinser.js

```diff
@@ -265,36 +271,55 @@ export async function createSlice(
     return { tempDir: sliceOutputDir, slicesFile };
   }
   console.log(
-    `Creating ${sliceType} slice for ${path.resolve(
-      filePath,
-    )}. Please wait ...`,
+    `Creating ${sliceType} slice for ${resolve(filePath)}. Please wait ...`,
   );
-  const atomFile = path.join(sliceOutputDir, "app.atom");
+  const atomFile = join(sliceOutputDir, "app.atom");
   let args = [sliceType];
   // Support for crypto slices aka CBOM
   if (sliceType === "reachables" && options.includeCrypto) {
     args.push("--include-crypto");
   }
-  if (sliceType === "usages" || ["ruby"].includes(language)) {
-    args.push("--remove-atom");
+  if (sliceType === "usages") {
+    // Generate OpenAPI specification for endpoints. Needs atom-tools pypi package to be installed.
+    args.push("--extract-endpoints");
+    if (process.env?.CDXGEN_IN_CONTAINER !== "true") {
+      console.log(
+        "Use an official cdxgen container image to improve the precision of endpoints detection (for SaaSBOM).",
+      );
+    }
+    if (["ruby", "scala"].includes(language)) {
+      args.push("--remove-atom");
+    }
+    if (["scala"].includes(language)) {
+      semanticsSlicesFile = join(
+        sliceOutputDir,
+        basename(options.semanticsSlicesFile || "semantics.slices.json"),
+      );
+    }
   }
   args = args.concat([
     "-l",
     language,
     "-o",
-    path.resolve(atomFile),
+    resolve(atomFile),
     "--slice-outfile",
-    path.resolve(slicesFile),
+    resolve(slicesFile),
   ]);
   // For projects with several layers, slice depth needs to be increased from the default 7 to 15 or 20
   // This would increase the time but would yield more deeper paths
   if (sliceType === "data-flow" && process.env.ATOM_SLICE_DEPTH) {
     args.push("--slice-depth");
     args.push(process.env.ATOM_SLICE_DEPTH);
   }
-
-  args.push(path.resolve(filePath));
-  const result = executeAtom(filePath, args);
+  args.push(resolve(filePath));
+  // Execute atom
+  const result = executeAtom(filePath, args, {
+    ATOM_TOOLS_OPENAPI_FILENAME: openapiSpecFile, // The file would get over-written
+    ATOM_TOOLS_OPENAPI_FORMAT:
+      process.env?.ATOM_TOOLS_OPENAPI_FORMAT || "openapi3.1.0", // editor.swagger.io doesn't support 3.1.0 yet
+    ATOM_TOOLS_WORK_DIR: resolve(filePath), // This must be the directory containing semantics.slices.json
+    OPENAPI_SERVER_URL: process.env?.OPENAPI_SERVER_URL,
```

Comment:
We need to add `./node_modules/.bin` to the PATH variable here to help atom find other tools such as `scalasem`.